### PR TITLE
png_loader: Added libpng based png loading

### PR DESCRIPTION
**Pull request by MathewP1**
- Description : Added loading png images using libpng. To enable this feature, change the `png` option in `meson_options` to **true**.
- Sample: #285

### CODE REVIEW

File: src/loaders/png/tvgPngLoader.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    png_bytep buffer;
+    image.format = PNG_FORMAT_BGRA;
+    buffer = static_cast<png_bytep>(malloc(PNG_IMAGE_SIZE(image)));
+    if (!buffer) {
+        // out of memory, only time when libpng doesnt free its data
+        png_image_free(&image);
+        return false;
+    }
+    if (!png_image_finish_read(&image, NULL, buffer, 0, NULL)) return false;
+    content = reinterpret_cast<uint32_t*>(buffer);
```
Comment (JSUYA) [inline src/loaders/png/tvgPngLoader.cpp:35]: I think this part can be moved to read().

Comment (MathewP1) [reply src/loaders/png/tvgPngLoader.cpp:35]: I will move it if you want me to, although I think that it doesn't really matter in this case.

Comment (JSUYA) [reply src/loaders/png/tvgPngLoader.cpp:35]: Hi 
You can declare `png_image image` as a member field of PngLoader class.
PngLoader inherits Task (tvgTaskScheduler.h) and can execute `buffer = malloc(image)` and `png_..._read()` in thread. please see tvgSvgLoader class.

File: src/loaders/png/tvgPngLoader.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#include <fstream>
+#include <string>
+#include "tvgLoaderMgr.h"
+#include "tvgPngLoader.h"
+// #include "png.h"
```
Comment (JSUYA) [inline src/loaders/png/tvgPngLoader.cpp:5]: please remove unnecessary line

**Comment by JSUYA** [comment]
Please write module and feature in the title of this PR
An appropriate title for this pr would be `png_loader PngLoader: ...`.

https://github.com/Samsung/thorvg/blob/master/CONTRIBUTING.md

File: meson_options.txt [hunk 3]
```diff
@@ -33,6 +33,11 @@ option('examples',
     value: false,
     description: 'Enable building examples')
 
+option('png',
```
Comment (JSUYA) [inline meson_options.txt:36]: I think the png option should be elements of the loaders' options.

Comment (hermet) [reply meson_options.txt:36]: @MathewP1  ping?

File: src/loaders/png/tvgPngLoader.cpp (hunk 2 above)
Comment (hermet) [reply src/loaders/png/tvgPngLoader.cpp:5]: @MathewP1 ping?

**Review by hermet — APPROVED** [review]

**Pull request merged**